feat: add `--fields` argument to `pixi list`

baszalmstra opened the pull request:
### Description

Adds the `--fields` argument to pixi list and exposes just about everything possible.

Fixes #5102
Fixes #4715 
Fixes #2076 
Fixes #1558

### How Has This Been Tested?

- [x] User tested

### AI Disclosure
<!--- Remove this section if your PR does not contain AI-generated content. --->
- [x] This PR contains AI-generated content.
  - [x] I have tested any AI-generated content in my PR.
  - [x] I take responsibility for any AI-generated content in my PR.
<!--- If you used AI to generate code, please specify the tool used and the prompt below. --->
Tools: Claude Opus 4.5

### Checklist:
<!--- Remove the non relevant items. --->
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added sufficient tests to cover my changes.
- [x] I have verified that changes that would impact the JSON schema have been made in `schema/model.py`.

bollwyvl commented:
Also, whilst mucking about in here: some care to alignment would be helpful: pretty much anything in an ordinal domain would benefit from being right-aligned. `size`, specifically, would also benefit from always including one trailing decimal place, as per https://github.com/prefix-dev/rattler-build/pull/1399

```
name           version        build                 size    kind   source
openssl        3.6.0          h26f9b46_0        3 MiB       conda  https://conda.anaconda.org/conda-forge/
oras-py        0.1.14         pyhd8ed1ab_0      33.6 KiB    conda  https://conda.anaconda.org/conda-forge/
packaging      25.0           pyh29332c3_1      61 KiB      conda  https://conda.anaconda.org/conda-forge/
```
vs:
```
name           version        build                 size    kind   source
openssl        3.6.0          h26f9b46_0         3.0 MiB    conda  https://conda.anaconda.org/conda-forge/
oras-py        0.1.14         pyhd8ed1ab_0      33.6 KiB    conda  https://conda.anaconda.org/conda-forge/
packaging      25.0           pyh29332c3_1      61.0 KiB    conda  https://conda.anaconda.org/conda-forge/
```

This still leaves size somewhat difficult to visually inspect: i believe `rattler-build` now makes "sizes over some threshhold" a different color, but this approach (or any bold/color) can get lost in logs. I'm not sure how to encode this, as my one [ASCII proposal](https://github.com/prefix-dev/rattler-build/pull/1399#issuecomment-2624552631) was not well-received.

Another example of data loss is the (useful) direct/transitive dependency status, with a direct dependency encoded as a bold, green package name. Perhaps a `depth` field (if easy) would be even more useful, especially if direct dependencies were omitted entirely.

```
depth   name           version        build                 size    kind   source
    4   openssl        3.6.0          h26f9b46_0         3.0 MiB    conda  https://conda.anaconda.org/conda-forge/
    3   oras-py        0.1.14         pyhd8ed1ab_0      33.6 KiB    conda  https://conda.anaconda.org/conda-forge/
        packaging      25.0           pyh29332c3_1      61.0 KiB    conda  https://conda.anaconda.org/conda-forge/
```

baszalmstra commented:
I added the alignment for `size`:

<img width="1836" height="393" alt="image" src="https://github.com/user-attachments/assets/3124e436-7546-402e-a72d-fe3edc37b026" />

Computing the depth is a bit more involved but I added a field for the `requested-spec`: 

<img width="829" height="781" alt="image" src="https://github.com/user-attachments/assets/49ed615a-4adc-48b4-b0b8-9c1e6935812f" />

bollwyvl commented:
The _Requested_ is great, and shows actual information: could get long for some matchspecs out there in the wild, but probably more useful than some tree-walking metric which i've frankly never seen anywhere else, certainly not worth the hustle to dig that up if it's not just lying around.

Again regarding space/noise: I'm not sure how much value the second decimal place provides, but hardly seems worth doing anything about, and certainly would be relevant for the `GiB` unit.

nichmor reviewed (approved):
Loooks good technically and semantically!
